### PR TITLE
Implement the Starlark index expression assignment operation

### PR DESCRIPTION
**Pull request by mahmoudimus**
## Description of changes in release / Impact of release:

In a nutshell, this PR allows one to set an object's value using `object[key] = value`.

[Starlark's specification](https://github.com/bazelbuild/starlark/blob/master/spec.md#index-expressions) actually supports this, but the [Java implementation](https://github.com/bazelbuild/starlark/issues/206) does not, even though the Go and Rust implementation do. 

This is an initial implementation attempt to do so. In addition, the failing tests are now passing to ensure that this behavior is additive and enhancing while maintaining backward compatibility. A nice side effect of this PR is that it also addresses https://github.com/verygoodsecurity/starlarky/issues/292. 

## Documentation

Read the `StarlarkSetIndexable` class and observe the implementation in `LarkyIndexable`

### Is this a breaking change?
- [ ] Yes
- [x] No

### Is there a way to disable the change?
- [x] Use previous release
- [ ] Use a feature flag
- [ ] No

### CODE REVIEW

File: larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java [hunk 1]
```diff
@@ -1,23 +1,252 @@
...
 import net.starlark.java.eval.Starlark;
-import net.starlark.java.eval.StarlarkMapping;
+import net.starlark.java.eval.StarlarkEvalWrapper;
+import net.starlark.java.eval.StarlarkIterable;
+import net.starlark.java.eval.StarlarkList;
 import net.starlark.java.eval.StarlarkSemantics;
 import net.starlark.java.eval.StarlarkThread;
+import net.starlark.java.eval.Tuple;
 
+import org.jetbrains.annotations.NotNull;
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java:27]: Prob use `NonNull`

File: larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java [hunk 2]
```diff
@@ -1,23 +1,252 @@
...
 
-public interface LarkyMapping<K, V> extends StarlarkMapping<K, V>, LarkyIndexable {
 
-  default Object get__setitem__() {
-    return getField(PyProtocols.__SETITEM__);
+public interface LarkyMapping<K, V> extends Map<K, V>, LarkyIndexable, Mutability.Freezable, StarlarkIterable<K> {
+
+  void freeze(); // this.mutability = Mutability.IMMUTABLE;
```
Comment (Iapetus999) [inline larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java:32]: ws, do these need JavaDoc and/or `@override`?

File: larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java
[the remark this answers is not in the document]
Comment (inssein) [reply larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java:1]: When I click on Load Diff it does seem to show a diff.

[the remark this answers is not in the document]
Comment (mahmoudimus) [reply larky/src/main/java/com/verygood/security/larky/modules/types/LarkyMapping.java:1]: There's no need for a StarlarkMapping and a LarkyMapping once this change is made. This removes the need for StarlarkMapping and lets us merge the file in LarkyMapping.

**Review by sharifelgamal — APPROVED** [review]

**Pull request merged**